ComponentKit/Core/CKComponent.h

```diff
@@ -55,4 +57,9 @@ struct CKComponentViewContext {
  */
 - (id)nextResponder;
 
+/**
+ This method composes an infra-driven component tree.
+ */
+- (std::vector<CKComponent *>)render:(id)state;
```

Comment:
Does this need to be on `CKComponent`? Could it instead be part of a separate subclass, à la `CKCompositeComponent`?

Reply:
The initializer of `CKCompositeComponent` takes a child component as a parameter. If we add the the render method only for `CKCompositeComponent`, how do we make sure that we create the child only in the render method and not in the constructor?

Reply:
I think @adamjernst  was suggesting to have a CKRenderEnabledComponent subclass

Reply:
What would be the benefit to have it on a separate subclass and not on `CKComponent`?

Reply:
I think I understand your point better now, we could have something like:
`CKComponent` would be a leaf component.
`CKChildComponent` would implement render that returns a child.
`CKFlexboxComponent` would be the "group" component, its render returns a vector of children.

In this case, we have to check in the infra the return type of the component from the render method and operate differently for each use case.